src/promptflow/promptflow/_sdk/_pf_client.py

```diff
@@ -125,15 +126,26 @@ def stream(self, run: Union[str, Run]) -> Run:
         """
         return self.runs.stream(run)
 
-    def get_details(self, run: Union[str, Run]) -> pd.DataFrame:
-        """Get run inputs and outputs.
+    def get_details(
```

Comment:
Ideally in SDK interface it should be a iterable API.

The max-results and all_results is only in CLI API

Reply:
Opened a task to do this:
[Task 2701793]: Revisit SDK API, may change to an iterable interface